syntax = "proto3";

service ModelLoader 
{
	// Server streaming RPC, as the models have many vertices = many data
	rpc GetModel (ModelRequest) returns (stream ModelReply) {}
	rpc GetModelSize (ModelRequest) returns (stream ModelReply) {}
}

message ModelRequest
{
  string modelName = 1;
}

message ModelReply
{
  // string or bytes for obj file?
  bytes objFile = 1;
  int32 texWidth = 2;
  int32 texHeight = 3;
  bytes texture = 4;
  int32 objFileSize = 5;
}